src/scala/microsoft-spark-2-4/src/main/scala/org/apache/spark/deploy/dotnet/DotnetRunner.scala

```diff
@@ -123,6 +125,17 @@ object DotnetRunner extends Logging {
       if (!runInDebugMode) {
         var returnCode = -1
         var process: Process = null
+        val enableLogRedirection: Boolean = sys.props
+            .getOrElse(
+              ERROR_REDIRECITON_ENABLED.key,
+              ERROR_REDIRECITON_ENABLED.defaultValue.get.toString)
+            .toBoolean
+        val circularBufferSize = sys.props
+            .getOrElse(
+              ERROR_BUFFER_SIZE.key,
+              ERROR_BUFFER_SIZE.defaultValue.get.toString).toInt
+        val stderrBuffer = new CircularBuffer(circularBufferSize)
```

Comment:
Shouldn't we be creating the `stderrBuffer` only if `enableLogRedirection` is enabled?